ddclient did not update the dynamic DNS entry if the config changes. 
This patch by Frans Pop to fixes this problem reported as  Debian bug 
#217041. 
Index: ddclient
===================================================================
--- ddclient.orig	2006-04-05 01:39:25.000000000 +0200
+++ ddclient	2006-04-05 01:39:37.000000000 +0200
@@ -1876,6 +1876,13 @@
     } elsif (defined($sub) && &$sub($host)) {
 	$update = 1;
 
+    } elsif (($cache{$host}{'static'} ne $config{$host}{'static'}) ||
+	     ($cache{$host}{'wildcard'} ne $config{$host}{'wildcard'}) ||
+	     ($cache{$host}{'mx'} ne $config{$host}{'mx'}) ||
+	     ($cache{$host}{'backupmx'} ne $config{$host}{'backupmx'})) {
+	info("updating %s because host settings have been changed.", $host);
+	$update = 1;
+
     } else {
 	success("%s: skipped: IP address was already set to %s.", $host, $ip)
 	    if opt('verbose');
